compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -47,6 +48,8 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
   private var myAnyClass: ClassSymbol = null
   private var myNothingClass: ClassSymbol = null
   private var myNullClass: ClassSymbol = null
+  private var myPhantomAnyClass: ClassSymbol = null
```

Comment:
My last comment only applies to `myPhantomAnyClass`, `myPhantomNothingClass` is used almost as frequently as `myNullClass`.